src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -80,6 +68,11 @@ public override async Task ExecuteCommandAsync()
 
             foreach (PackageSource source in listEndpoints)
             {
+                if (source.IsHttp && !source.IsHttps)
+                {
+                    logger.LogWarning(string.Format(CultureInfo.CurrentCulture, LocalizedResourceManager.GetString("Warning_HttpServerUsage"), "search", source.Source));
+                }
+
```

Comment:
Can we add a single warning here instead of warning per source? For example, I see list command code has `WarnForHTTPSources` utility method which adds single warning for all HTTP sources